lib/internal/http2/core.js

```diff
@@ -3098,7 +3098,7 @@ function initializeTLSOptions(options, servername) {
   options.ALPNProtocols = ['h2'];
   if (options.allowHTTP1 === true)
     ArrayPrototypePush(options.ALPNProtocols, 'http/1.1');
-  if (servername !== undefined && options.servername === undefined)
+  if (servername !== undefined && !options.servername)
```

Comment:
```suggestion
  if (options.servername)
```

Reply:
how would this help?

Reply:
The `options.servername` already has the value we provide as a parameter, it will be overrided if it's not truthy.

Reply:
How does this help? Can you explain??

Reply:
Ah sorry for confusion, the first `servername` is not `options.servername`. My bad.

Reply:
It's okay. Happens.

Reply:
@szmarczak , can we resolve this? Since `initializeTLSOptions` is a function, it should take care of validating the parameters on it's own, removing `servername !== undefined` will make the function depend on the caller to send the proper value, even if it's `undefined`, we simply assign `undefined` with `undefined`.

Reply:
[I don't see a button to resolve this](https://user-images.githubusercontent.com/36894700/131485647-2b8d8347-1f42-4cd2-ad56-cecc2288cd82.png). Maybe a GitHub bug, dunno.